avocado/core/job.py

```diff
@@ -462,6 +465,7 @@ def _run(self):
         """
         self._setup_job_results()
         self.__start_job_logging()
+        self.job_pre_post_dispatcher.map_methods('pre', self)
```

Comment:
I'm still not 100% sure this is the right location. It depends on how we define the pre-job task.

1. (current) run as early as the test results are created, before the url discovery and multiplexation happens.
2. run when option validation is complete (therefor after test discovery and multiplexation)

The difference is, that currently first pre-job plugins are executed, therefor the user sees their output and after that the job log occurs:

```
avocado run passtest --show-job-log
Running '/etc/avocado/scripts/job/pre.d/sleep'
[stderr] + sleep 2
[stderr] + exit -1
Pre job script "/etc/avocado/scripts/job/pre.d/sleep" exited with status "255"
Commands configured by file: /etc/avocado/sysinfo/commands
Files configured by file: /etc/avocado/sysinfo/files
Profilers configured by file: /etc/avocado/sysinfo/profilers
Profilers declared: ['vmstat 1', 'journalctl -f']
Profiler disabled
Command line: /usr/bin/avocado run passtest --show-job-log

Avocado version: 0.34.0
Avocado git repo info
Top commit: e85e65e7c991383d68965562421bcbf3cc16a82f
Branch: HEAD
```
which might be a bit confusing. On the other hand it allows one to prepare the environment for test discovery (check if test repository is updated, fetch new changes and then run the discovery).

The later would not allow this, but the logs would be more like we know it, pre-jobs would not be executed in case of incorrect cmdline.

I'm inclined towards the current implementation, but we should file a bug to log the job headline earlier (the same problem is eg. with avocado-vt, which also prints some avocado-vt information before the avocado job headline).

Note by headline I mean:

```
Commands configured by file: /etc/avocado/sysinfo/commands
Files configured by file: /etc/avocado/sysinfo/files
Profilers configured by file: /etc/avocado/sysinfo/profilers
Profilers declared: ['vmstat 1', 'journalctl -f']
Profiler disabled
Command line: /usr/bin/avocado run passtest --show-job-log

Avocado version: 0.34.0
Avocado git repo info
Top commit: e85e65e7c991383d68965562421bcbf3cc16a82f
Branch: HEAD
```

Or at least:

```
Command line: /usr/bin/avocado run passtest --show-job-log
```